Lib/pydoc.py

```diff
@@ -1669,9 +1669,11 @@ def render_doc(thing, title='Python Library Documentation: %s', forceload=0,
     return title % desc + '\n\n' + renderer.document(object, name)
 
 def doc(thing, title='Python Library Documentation: %s', forceload=0,
-        output=None):
+        output=None, follow_wrapped=True):
     """Display text documentation, given an object or a path to an object."""
     try:
+        if not follow_wrapped:
+            thing = inspect.unwrap(thing)
```

Comment:
Perhaps this should stop when finding an object  with a `__doc__` attribute?
```suggestion
            thing = inspect.unwrap(thing, stop=lambda func: getattr(func, '__doc__', None)
```